File: character_profile_generator.java
import java.util.Scanner;

/**
 * Character Profile Generator
 *
 * This program collects information about a user's custom character,
 * including name, race, class, and several numeric stats. It demonstrates
 * the use of Java's Scanner class, handling both String and numeric input,
 * and properly managing the interaction between nextLine() and nextInt().
 *
 * The program also performs simple arithmetic by calculating an overall
 * power level based on the character's stats. It uses escape sequences
 * for clean formatting and organizes the logic into clear input,
 * processing, and output sections.
 * 
 * @author Jonathan
 */


public class CharacterProfileGenerator {

	public static void main(String[] args) {
		
		String charName;
		String charRace;
		String charClass;
		int strength;
		int health;
		int speed;
		double powerLevel;
		Scanner stdin;
		
		System.out.println("--------------------------------");
		System.out.println("   Character Profile Summary	");
		System.out.println("--------------------------------");
		
		
		stdin = new Scanner(System.in);
		System.out.print("Enter your character's name: ");
		charName = stdin.nextLine();
		System.out.print("Enter your character's race: ");
		charRace = stdin.nextLine();
		System.out.print("Enter your character's class: ");
		charClass = stdin.nextLine();
		
		System.out.print("\nEnter Strength (0-100): ");
		strength = stdin.nextInt();
		System.out.print("Enter Health (0-100): ");
		health = stdin.nextInt();
		System.out.print("Enter Speed (0-100): ");
		speed = stdin.nextInt();
		stdin.close();
		
		System.out.println("\n--------------------------------");
		System.out.println("     Final Character Sheet	");
		System.out.println("--------------------------------");
		
		powerLevel = (strength + health + speed) / 3;
		
		System.out.println("\nName: " + charName);
		System.out.println("Race: " + charRace);
		System.out.println("Class: " + charClass);
		
		System.out.println("\nStats:");
		System.out.println("\tStrength: " + strength);
		System.out.println("\tHealth: " + health);
		System.out.println("\tSpeed: " + speed);
		
		System.out.println("\nOverall Power Level: " + powerLevel);
		System.out.println("\n'May " + charName + "'s arrows fly true.'");
		System.out.println("--------------------------------");
		

	}

}

File: character_profile_generator.py
"""
    This program collects information about a user's custom character,
    including name, race, class, and several numeric stats. This program
    also performs simple math by calculating an overall power level
    based on the character's stats.
"""

# Annotate variables
charName: str
charRace: str
charClass: str
strength: int
health: int
speed: int
powerLevel: float

print("--------------------------------")
print("   Character Profile Summary   ")
print("--------------------------------")

# Ask user for character info
charName = input("Enter your character's name: ")
charRace = input("Enter your character's race: ")
charClass = input("Enter your character's class: ")

strength = int(input("\nEnter Strength (0-100): "))
health = int(input("Enter Health (0-100): "))
speed = int(input("Enter Spees (0-100): "))

# Display output
print("--------------------------------")
print("     Final character Sheet     ")
print("--------------------------------")

powerLevel = (strength + health + speed) / 3

print(f"\nName: {charName}")
print(f"Race: {charRace}")
print(f"Class: {charClass}")

print("\nStats:")
print(f"\tStrength: {strength}")
print(f"\tHealth: {health}")
print(f"\tSpeed: {speed}")

print(f"\nOverall Power Level: {powerLevel:.2f}")
print(f"\n'May {charName}'s arrows fly true.'")
print("--------------------------------")
